inference/core/workflows/core_steps/models/foundation/cogvlm.py

```diff
@@ -0,0 +1,344 @@
+            prompt=prompt,
+            model_manager=self._model_manager,
+            api_key=self._api_key,
+        )
+        structured_output = turn_raw_lmm_output_into_structured(
+            raw_output=raw_output,
+            expected_output=json_output,
+        )
+        predictions = [
+            {
+                "raw_output": raw["content"],
+                "image": raw["image"],
+                "structured_output": structured,
+                **structured,
+            }
+            for raw, structured in zip(raw_output, structured_output)
+        ]
+        for prediction, image in zip(predictions, images):
+            prediction[PARENT_ID_KEY] = image.parent_metadata.parent_id
+            prediction[ROOT_PARENT_ID_KEY] = (
+                image.workflow_root_ancestor_metadata.parent_id
+            )
+        return predictions
+
+    async def run_remotely(
+        self,
+        images: Batch[WorkflowImageData],
+        prompt: str,
+        json_output: Optional[Dict[str, str]],
+    ) -> BlockResult:
+        if json_output:
+            prompt = (
+                f"{prompt}\n\nVALID response format is JSON:\n"
+                f"{json.dumps(json_output, indent=4)}"
+            )
+        inference_images = [i.to_inference_format() for i in images]
+        raw_output = await get_cogvlm_generations_from_remote_api(
+            image=inference_images,
+            prompt=prompt,
+            api_key=self._api_key,
+        )
+        structured_output = turn_raw_lmm_output_into_structured(
+            raw_output=raw_output,
+            expected_output=json_output,
+        )
+        predictions = [
+            {
+                "raw_output": raw["content"],
+                "image": raw["image"],
+                "structured_output": structured,
+                **structured,
+            }
+            for raw, structured in zip(raw_output, structured_output)
+        ]
+        for prediction, image in zip(predictions, images):
+            prediction[PARENT_ID_KEY] = image.parent_metadata.parent_id
+            prediction[ROOT_PARENT_ID_KEY] = (
+                image.workflow_root_ancestor_metadata.parent_id
+            )
+        return predictions
+
+
+async def get_cogvlm_generations_locally(
+    image: List[dict],
+    prompt: str,
+    model_manager: ModelManager,
+    api_key: Optional[str],
+) -> List[Dict[str, Any]]:
+    serialised_result = []
+    for single_image in image:
+        loaded_image, _ = load_image(single_image)
+        image_metadata = {
+            "width": loaded_image.shape[1],
+            "height": loaded_image.shape[0],
+        }
+        inference_request = CogVLMInferenceRequest(
+            image=single_image,
+            prompt=prompt,
+            api_key=api_key,
+        )
+        model_id = load_core_model(
+            model_manager=model_manager,
+            inference_request=inference_request,
+            core_model="cogvlm",
```

Comment:
This is just a thought, no action required - we could potentially parametrize this string and pass it through block init with `"cogvlm"` default